### DISCUSSION TITLE
use of examples already built on arduino UNO

### DISCUSSION THREAD

**Discussion by Antonin-m84** [stm32duino/Arduino_Core_STM32#2370] [Q&A]
Hello ! 

I am trying to use the 2 examples provided for the ToF VL53L8CX sensor with the associated libraries but I am encountering many problems. 
The Arduino IDE connected to my arduino UNO encounters lots of errors, I try to modify the .ino but an AI (cursor tutor - ChatGPT) tells me that the problem comes from the code of the files in the "src" folder. 
I don't know what to do, should I change so many things? What am I doing wrong?

https://github.com/stm32duino/VL53L8CX

**Comment by fpistm** [comment]
Hello @Antonin-m84 
@cparata could you answer to @Antonin-m84, please?

**Comment by cparata** [comment]
Hello @Antonin-m84 ,
Which Arduino Uno are you using? R3 or R4? I think that the Arduino Uno R3 does not have enough memory to run the VL53L8CX library. The Arduino Uno R4 instead should be fine.
Best Regards,
Carlo

**Reply by Antonin-m84** [reply]
Arduino Uno R3 SMD :/ 
There is no chance, no way to make it work? 
Thx

**Reply by cparata** [reply]
Unfortunately not. If you check the source code in the vl53l8cx_buffers.h file, you will see that the "VL53L8CX_FIRMWARE" buffer is too big in size to be read inside the flash memory of the Arduino UNO R3 (the variable has about a 86KB size versus the 32KB of the flash size available in the Arduino Uno R3). The Arduino Uno R4 instead has 256KB of flash size, so it should be fine.
Best Regards,
Carlo

**Comment by Antonin-m84** [comment]
Hi,

What do you think of this STM32? Does it have enough flash memory? I wouldn't have any problems with it? 
https://www.mouser.fr/ProductDetail/STMicroelectronics/NUCLEO-F401RE?qs=sGAEpiMZZMuqBwn8WqcFUv%2FX0DKhApUpi46qP7WpjrffIid8Wo1rTg%3D%3D

I saw that it had been used to develop the "Ultra Lite Driver" see screenshot
![image](https://github.com/stm32duino/Arduino_Core_STM32/assets/105228978/5b3173bb-56b2-46a9-a5f8-2ff69a99cf9a)
Thx a lot
Antonin

**Comment by cparata** [comment]
Hello @Antonin-m84 ,
you can use for sure the NUCLEO-F401RE. I tested the VL53L8CX library with this board, so the library works fine with that Nucleo board.
Best Regards,
Carlo
